x/evmutil/keeper/msg_server.go

```diff
@@ -102,3 +106,66 @@ func (s msgServer) ConvertERC20ToCoin(
 
 	return &types.MsgConvertERC20ToCoinResponse{}, nil
 }
+
+// MsgEVMCall executes the msg data again the go-ethereum EVM.
+func (k msgServer) EVMCall(goCtx context.Context, msg *types.MsgEVMCall) (*types.MsgEVMCallResponse, error) {
+	expMsgAuthority := k.keeper.GetAuthority()
+	if expMsgAuthority != msg.Authority {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidSigner, "invalid authority; expected %s, got %s", expMsgAuthority, msg.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	toAddr, err := types.NewInternalEVMAddressFromString(msg.To)
+	if err != nil {
+		return nil, fmt.Errorf("invalid to evm address: %w", err)
+	}
+
+	// convert empty data to valid hex string
+	msgData := msg.Data
+	if len(msg.Data) == 0 {
+		msgData = "0x"
+	}
+
+	data, err := hexutil.Decode(msgData)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode msg data: %w", err)
+	}
+	authorityAddr, err := sdk.AccAddressFromBech32(expMsgAuthority)
+	if err != nil {
+		return nil, fmt.Errorf("invalid authority address: %w", err)
+	}
+	authorityEvmAddr := common.BytesToAddress(authorityAddr.Bytes())
+
+	amt := msg.Amount.BigInt()
+	if amt == nil {
+		amt = big.NewInt(0)
+	}
+	_, err = k.keeper.CallEVMWithData(
+		ctx,
+		authorityEvmAddr,
+		&toAddr,
+		data,
+		amt,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("evm call failed: %w", err)
```

Comment:
for troubleshooting might be useful to debug log the params passed to `CallEVMWithData` above here

Reply:
👍